--- RouteMatriXRandomizer/092/data/multiworld/boot.lua	2025-08-16 11:15:37.000000000 +0800
+++ src/092/boot.lua	2025-09-02 14:12:09.223147148 +0800
@@ -2,12 +2,13 @@
 	error("BizHawk emu ( + Faust Snes core ) only.")
 end
 
-package.path = package.path..";lua\\?.lua"
+package.path = package.path..";lua/?.lua"
 require "lib_EmuWrapper"
 require "lib_text"
 require "lib_text_j"
 require "itemName"
 require "lib_input2"
+require "./progress_tracker/progress_tracker" -- by fsworld009
 
 cpu_ = cpu
 cpu = ew.cpu
@@ -52,10 +53,10 @@
 local addrDefaultItems = {0xBFFDE0,0xBFFDE0,0xCFFDE0,}
 
 -------------------- 定数 --------------------
-local filename_boot = ".\\boot.smc"
-local filename_game = {".\\bin\\x1.smc",".\\bin\\x2.smc",".\\bin\\x3.smc"}
-local filenameSave = ".\\save.txt"
-local filenameSavestateCoreInfo = ".\\savestateInfo.txt"
+local filename_boot = "./boot.smc"
+local filename_game = {"./bin/x1.smc","./bin/x2.smc","./bin/x3.smc"}
+local filenameSave = "./save.txt"
+local filenameSavestateCoreInfo = "./savestateInfo.txt"
 
 local cMS_switchGame = 0x00
 local cMS_switchGameL = 0x01
@@ -135,7 +136,7 @@
 end
 
 local function getSaveStateFilename(title)
-	return ".\\savestate_x" .. title .. ".State"
+	return "./savestate_x" .. title .. ".State"
 end
 
 
@@ -266,6 +267,10 @@
 			quickInfo("Saved.",30)
 		end
 	end
+
+	if forceUpdate or updated or atBoot then
+		progress_tracker.onSyncProgress(sessionInfo, sessionSave) -- by fsworld009
+	end
 end
 
 --sessionInfo.currentGameのゲームをロードする
@@ -391,6 +396,7 @@
 	if sessionInfo.acquiredTimer < maxFrames - minFrames and #sessionInfo.acquiredItemQueue ~= 0 then
 		sessionInfo.acquiredItem = sessionInfo.acquiredItemQueue[1]
 		table.remove( sessionInfo.acquiredItemQueue , 1 )
+		progress_tracker.onAcquireItem(sessionInfo, sessionSave)  -- by fsworld009
 		sessionInfo.acquiredTimer = maxFrames
 	end
 	if sessionInfo.acquiredItem then
